common/token-module/src/lib.rs

```diff
@@ -110,16 +138,47 @@ pub trait TokenModule: fee_estimator_module::FeeEstimatorModule {
         roles.has_role(role)
     }
 
+    #[only_owner]
+    #[endpoint(setMultiTransferContractAddress)]
+    fn set_multi_transfer_contract_address(&self, opt_new_address: OptionalValue<ManagedAddress>) {
```

Comment:
Not sure you need all these extra checks. This is an owner_only endpoint, so only multisig SC can call this. Furthermore, in multisig you read it from the storage, where you previously set it at init (where you already had check for the address to be a SC).

The same thing for the `setEsdtSafeContractAddress` endpoint in `multi-transfer-esdt` SC.